src/Components/test/testassets/BasicTestApp/RouterTest/TestRouterWithDynamicAssembly.razor

```diff
@@ -0,0 +1,56 @@
+@using Microsoft.AspNetCore.Components.Routing
+@using Microsoft.AspNetCore.Components.WebAssembly
+@using System.Linq
+@using System.Reflection
+
+@inject WebAssemblyDynamicAssemblyLoader lazyLoader
+
+<Router AppAssembly="@typeof(BasicTestApp.Program).Assembly" AdditionalAssemblies="@lazyLoadedAssemblies" OnNavigate="@OnNavigate">
+    <Found Context="routeData">
+        <RouteView RouteData="@routeData" />
+    </Found>
+    <NotFound>
+        <LayoutView Layout="@typeof(RouterTestLayout)">
+            <div id="test-info">Oops, that component wasn't found!</div>
+        </LayoutView>
+    </NotFound>
+</Router>
+
+@code {
+    private List<Assembly> lazyLoadedAssemblies = new List<Assembly>();
+
+    public bool OnNavigate (string uri)
+    {
+        Console.WriteLine($"Running OnNavigate for {uri}...");
+        _ = LoadAssemblies(uri);
+        return false;
+    }
+
+    private async Task LoadAssemblies(string uri)
+    {
+        bool loadedAssemblies = false;
+        try
+        {
+            if (uri.EndsWith("WithDynamicAssembly"))
+            {
+                loadedAssemblies = true;
```

Comment:
You have this variable here, but you don't do nothing with it.

Reply:
It might be nice if you use it to put something on the UI

Reply:
It's used in the `finally` clause of this piece of code. We only call `StateHasChanged` if assemblies needed to be loaded on the present page.